src/main/java/com/leverx/learningmanagementsystem/db/service/dbmigrator/DataBaseMigrator.java

```diff
@@ -1,32 +1,26 @@
-package com.leverx.learningmanagementsystem.core.db.service;
+package com.leverx.learningmanagementsystem.db.service.dbmigrator;
 
 import com.leverx.learningmanagementsystem.core.exception.model.CustomLiquibaseException;
+import com.leverx.learningmanagementsystem.core.security.context.TenantContext;
+import com.leverx.learningmanagementsystem.multitenancy.connectionprovider.CustomMultiTenantConnectionProvider;
 import liquibase.Contexts;
 import liquibase.LabelExpression;
 import liquibase.Liquibase;
 import liquibase.database.Database;
 import liquibase.database.DatabaseFactory;
 import liquibase.database.jvm.JdbcConnection;
 import liquibase.resource.ClassLoaderResourceAccessor;
-import lombok.AllArgsConstructor;
-import lombok.extern.slf4j.Slf4j;
-import org.springframework.stereotype.Component;
 
-import javax.sql.DataSource;
 import java.sql.Connection;
 
-import static com.leverx.learningmanagementsystem.multitenancy.utils.MigrationUtils.DB_CHANGELOG;
+import static com.leverx.learningmanagementsystem.db.constants.DatabaseConstants.DB_CHANGELOG;
+import static com.leverx.learningmanagementsystem.db.constants.DatabaseConstants.SCHEMA;
 
-@Component
-@Slf4j
-@AllArgsConstructor
-public class DatabaseMigrator {
+public interface DataBaseMigrator {
 
-    private final DataSource dataSource;
-
-    public void migrateSchema(String schemaName) {
-        try (Connection connection = dataSource.getConnection()) {
-            connection.setSchema(schemaName);
+    default void migrateSchema(CustomMultiTenantConnectionProvider connectionProvider) {
+        String tenantId = TenantContext.getTenantId();
+        try (Connection connection = connectionProvider.getConnection(tenantId)) {
```

Comment:
this code breaks purpose of connections pool :(
try with resources closes provided resource in the end. For example your datasources has 10 connection, but after this method execution, you will have 9 as 1 was closed. You don't need to close connection in the end